Introduce `pylock.toml` (#899)
fandango-fuzzer/fandango: docs/Contributing.md

The [tests](sec:running-tests) are not run as part of pre-commit, because they take significant time to complete. Run them manually.
```

To support reproducible environments and compatibility across a wider range of tools in the Python ecosystem, we keep both `uv.lock` and `pylock.toml` up to date.

> - `uv.lock` is the lockfile generated by `uv` and captures the actual dependency set used in this project (i.e., source of truth).

Reviewer: ```suggestion
- `uv.lock` is the lockfile generated by `uv` and captures the actual dependency set used in this project (i.e., source of truth). Update with `uv lock`
```

> - `pylock.toml` is derived from `uv.lock` to support a wider range of ecosystem tools.

Reviewer: ```suggestion
- `pylock.toml` is derived from `uv.lock` to support a wider range of ecosystem tools. Update with `uv export --all-extras --locked --output-file=pylock.toml`.
```

> After updating dependencies in `pyproject.toml`, make sure to regenerate `uv.lock` with `uv lock` and update `pylock.toml` accordingly.

Reviewer: ```suggestion
After updating dependencies in `pyproject.toml`, make sure to regenerate `uv.lock` and `pylock.toml` to keep them up to date. Both pre-commit hooks and the CI pipeline contain checks for these.
```
